gammapy/spectrum/spectrum_dataset.py

```diff
@@ -0,0 +1,205 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from pathlib import Path
+import numpy as np
+from astropy.units import Quantity
+from ..utils.scripts import make_path
+from ..irf import EffectiveAreaTable, EnergyDispersion
+from .core import PHACountsSpectrum
+from .utils import SpectrumEvaluator
+from ..utils.fitting import Dataset, Parameters
+from..stats import wstat
+
+__all__ = [
+    "ONOFFSpectrumDataset"
+]
+
+
+class ONOFFSpectrumDataset(Dataset):
```

Comment:
In the current naming scheme we have have for Gammapy we rather use suffixes and PEP 8 recommends camel- case for class names. So I'd prefer `SpectrumDatasetOnOff` as a name. I actually also liked the `SpectrumDatasetOGIP` name from the prototype notebook, because the data structure is much closer to the OGIP format, with the `aeff` and `livetime` parameterisation.

Reply:
OK. Changed.

I have used the `ONOFF` terminology precisely because I did not want to confuse the `Dataset` with a convention for storing spectra. Of course, it is a bit more than a data format since there is, e.g., the prescription that IRFs are described as `arf`, `edisp`, `livetime` rather than e.g. `exposure` and `edisp`. Also we don't store a `alpha` parameter but deduce it from `ON.backscal` and `OFF.backscal`.

Conversely, you can have an `OGIP` compliant spectrum that has no `OFF` spectrum, but rather a model for background. In a sense, the current `SpectrumDataset` could also be stored in `OGIP`-compliant files. Also, not all `OGIP` files would be fitted with `wstat`.

So I have mixed views regarding the best approach here. Any opinion? @adonath

Reply:
I'm not 100% sure either what's best to do here. However I think the current solution, i.e. having the OGIP compliant data model and using it for our analyses is not a bad compromise. In general I'd prefer the data model based on `alpha` and `exposure`, just because it's more common in Gamma astronomy. But the only solution I see would be to introduce both `SpectrumDataset` and `SpectrumDatasetOGIP` as well as `SpectrumDatasetOnOff` and `SpectrumDatasetsOGIPOnOff`, which comes at the cost of some code duplication and proliferation of spectral dataset classes.